apps/elixir_chatbot_core/lib/documentation_manager/documentation_fragment.ex

```diff
@@ -73,7 +73,7 @@ defmodule ElixirChatbotCore.DocumentationManager.DocumentationFragment do
       nil,
       fn cur, acc ->
         heading_text =
-          if String.starts_with?(cur, String.duplicate("#", current_depth) <> " ") do
+          if not String.starts_with?(cur, "## Example") and String.starts_with?(cur, String.duplicate("#", current_depth) <> " ") do
```

Comment:
This is not terribly elegant, probably should replace it with some regex or linear search, but let's leave it as is for now

Reply:
I'm leaving it as it is but I've added a comment that it needs refactoring here.